test/WebJobs.Script.Tests.Integration/ApplicationInsights/ApplicationInsightsTestFixture.cs

```diff
@@ -33,7 +35,19 @@ public ApplicationInsightsTestFixture(string scriptRoot, string testId)
                 .UseStartup<Startup>()
                 .ConfigureServices(services =>
                 {
-                    ScriptSettingsManager.Instance.ApplicationInsightsInstrumentationKey = TestChannelLoggerProviderFactory.ApplicationInsightsKey;
+                    var settingsManager = new ScriptSettingsManager();
+                    settingsManager.SetConfigurationFactory(() =>
+                    {
+                        return new ConfigurationBuilder()
+                            .AddInMemoryCollection(new Dictionary<string, string>
+                            {
+                                [EnvironmentSettingNames.AppInsightsInstrumentationKey] = TestChannelLoggerProviderFactory.ApplicationInsightsKey
+                            })
+                            .AddEnvironmentVariables()
+                            .Build();
+                    });
```

Comment:
@mathewc -- this is why these tests were failing for you. I wasn't wiring up the setting correctly but it was falling back to my local environment variables. Now it sets it explicitly.